packages/credential-provider-process/src/index.ts

```diff
@@ -0,0 +1,144 @@
+            return await execPromise(credentialProcess)
+                .then((processResult: any) => {
+                    let data;
+                    try {
+                        data = JSON.parse(processResult);
+                    } catch {
+                        throw Error(`Profile ${profileName} credential_process returned invalid JSON.`)
+                    }
+
+                    const {
+                        Version: version,
+                        AccessKeyId: accessKeyId,
+                        SecretAccessKey: secretAccessKey,
+                        SessionToken: sessionToken,
+                        Expiration: expiration
+                    } = data;
+
+
+                    if (version !== 1) {
+                        throw Error(`Profile ${profileName} credential_process did not return Version 1.`)
+                    }
+                    
+                    if (accessKeyId === undefined || secretAccessKey === undefined) {
+                        throw Error(`Profile ${profileName} credential_process returned invalid credentials.`)
+                    }
+
+                    if (expiration) {
+                        const currentTime = new Date();
+                        const expireTime = new Date(expiration);
+                        if (expireTime < currentTime) {
+                            throw Error(`Profile ${profileName} credential_process returned expired credentials.`)
+                        }
+                        return {
+                            accessKeyId,
+                            secretAccessKey,
+                            sessionToken,
+                            expiration: Math.floor(
+                                (new Date(expiration)).valueOf() / 1000
+                            ),
+                        }
+                    } else {
+                        return {
+                            accessKeyId,
+                            secretAccessKey,
+                            sessionToken
+                        }
+                    }
+                }).catch((error: Error) => {
+                    throw new ProviderError(
+                        error.message
+                    )
+                });
+        } else {
+            throw new ProviderError(
+                `Profile ${profileName} did not contain credential_process.`
+            );
+        }
+    } else {
+        // If the profile cannot be parsed or does not contain the default or
+        // specified profile throw an error. This should be considered a terminal
+        // resolution error if a profile has been specified by the user (whether via
+        // a parameter, anenvironment variable, or another profile's `source_profile` key).
+        throw new ProviderError(
+            `Profile ${profileName} could not be found in shared credentials file.`
+        );
+    }
+}
+
+function parseKnownFiles(init: FromProcessInit): Promise<ParsedIniData> {
```

Comment:
Is this function the same that that in the shared ini loader? If so you can import the function from there. Since this package already depend on that package, it should change the dependency.

Reply:
Those functions are in credential-provider-ini, but aren't exported.

Reply:
It's fine to export those functions from `credential-provider-ini`